__tests__/app.test.js

```diff
@@ -221,6 +282,56 @@ describe('Error handling tests', () => {
                         expect(msg).toBe('Invalid Request')
                     })
             })
+            test('POST - /comments - responds responds with 400 error code and error message if requested article ID was invalid', () => {
+                return request(app)
+                    .post('/api/articles/art5/comments')
+                    .send({
+                        "username": "lurker",
+                        "body": "cool article"
+                    })
+                    .expect(400)
+                    .then(({ body: { msg } }) => {
+                        expect(msg).toBe('Invalid Request')
+                    })
+            })
+        })
+    })
+    describe('/api/articles/:article_id/comments', () => {
+        describe('POST - status 400 - invalid request body', () => {
+            test('responds with 400 erroor code and error message if provided body of the request does not contain username and/or body property value', () => {
+                return request(app)
+                    .post('/api/articles/3/comments')
+                    .send({
+                        "username": "lurker",
+                        "body": null
+                    })
+                    .expect(400)
+                    .then(({ body: { msg: noBodyMsg } }) => {
+                        return request(app)
+                            .post('/api/articles/3/comments')
+                            .send({
+                                "username": null,
+                                "body": "cool article"
+                            })
+                            .expect(400)
+                            .then(({ body: { msg: noUserMsg } }) => {
+                                expect(noBodyMsg).toBe('Invalid Request')
+                                expect(noUserMsg).toBe('Invalid Request')
+                            })
+                    })
+            })
+            test('responds with 404 error code and error message if provided username does not exists', () => {
+                return request(app)
+                    .post('/api/articles/3/comments')
+                    .send({
+                        "username": "deletedAccount",
+                        "body": "cool article"
+                    })
+                    .expect(404)
+                    .then(({ body: { msg } }) => {
+                        expect(msg).toBe('Not Found')
+                    })
+            })
```

Comment:
Really nice testing so far Alicja! Could you just add one more 201 test please, which ignores unnecessary properties? So if for example we add something apart from username and body into the object we're sending, it should ignore it and still post. 👍

Reply:
Thanks! Just added the test. Does it have everything it needs?